cloudant.js

```diff
@@ -47,15 +50,51 @@ function Cloudant(options, callback) {
     options.https = false;
   }
 
-  debug('Creating Cloudant client with options: %j', options);
-  var cloudantClient = new Client(options);
-  var cloudantRequest = function(req, callback) {
-    return cloudantClient.request(req, callback);
+  var cloudantClient, cloudantRequest;
+  var requestDefaults = {jar: false}; // default
+
+  if (typeof options.plugin === 'function') {
+    // legacy code path
+    debug('Using legacy custom plugin');
+    cloudantRequest = options.plugin;
+
+    if (options.requestDefaults) {
+      requestDefaults = options.requestDefaults;
+    } else {
+      requestDefaults = { gzip: true, jar: false };
+    }
+
+    // set library UA header
```

Comment:
Do we have two copies of this now?

Reply:
Yeah, this is a legacy code path. If the client specifies a legacy plugin then we avoid initialising the Cloudant client and instead substitute in their plugin. We still need to add the UA and setup the default agent config.

Reply:
It'd be nice if we could at least use the same agent/constants in both cases, but if it's not easy to put the code somewhere common then I'm ok with this.